### PR TITLE
infoschema, http: add two HTTP API for keyvis

### LINKED ISSUE #54281: When many tables under a DB, dashboard keyvis feature will cost a lot of resource
## Enhancement
img_v3_02c6_a75c806e-3a46-40ba-bdad-3cc752158b0g

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #54281

Problem Summary:

### What changed and how does it work?

see tidb_http_api.md

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

cluster has a 488k tables' database

before & after

image

before (10s CPU profile)

img_v3_02cr_f3c3eeed-f276-44bf-9651-6cbd064e37cg

after (10s CPU profile)

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: docs/tidb_http_api.md [hunk 1]
```diff
@@ -165,12 +165,34 @@
     curl http://{TiDBIP}:10080/schema/{db}
     ```
 
+    ```shell
+    curl http://{TiDBIP}:10080/schema/{db}?id_name_only=true
```
Comment (tangenta) [inline docs/tidb_http_api.md:169]: `id_name_only=true` seems not RESTFUL...

Comment (lance6716) [reply docs/tidb_http_api.md:169]: https://stackoverflow.com/questions/39979533/how-should-rest-api-accept-boolean-values here suggested that use `true` to represent boolean arguments

Comment (tangenta) [reply docs/tidb_http_api.md:169]: I prefer the design like `schema/{db}?fields=id,name`, but it is OK for now since keyvis is the only user.

**Review by tangenta — APPROVED** [review]

**Pull request merged**